dallinger/command_line/ec2.py

```diff
@@ -0,0 +1,233 @@
+@click.pass_context
+def list(ctx):
+    pass
+
+
+@list.command("instances")
+@click.option("--region", default=None, help="Region name")
+@click.option("--running", is_flag=True, help="List running instances")
+@click.option("--stopped", is_flag=True, help="List stopped instances")
+@click.option("--terminated", is_flag=True, help="List terminated instances")
+@click.option("--pem", default=None, help="Name of the PEM file to use")
+@click.pass_context
+def list__instances(ctx, region, running, stopped, terminated, pem):
+    filtered_states = []
+    if running:
+        filtered_states.append("running")
+    if stopped:
+        filtered_states.append("stopped")
+    if terminated:
+        filtered_states.append("terminated")
+    list_instances(region, filtered_states=filtered_states, pem=pem)
+
+
+@list.command("regions")
+@click.pass_context
+def list__regions(ctx):
+    list_regions()
+
+
+@list.command("instance_types")
+@click.option("--region", default=None, help="Region name")
+@click.pass_context
+def list__instance_types(ctx, region):
+    list_instance_types(region)
+
+
+@ec2.command("provision")
+@click.option("--name", required=True, help="Instance name")
+@click.option("--region", default=None, help="Region name")
+@click.option("--type", default="m5.xlarge", help="Instance type")
+@click.option("--storage", default=32, type=int, help="Storage in GB; default is 32 GB")
+@click.option(
+    "--pem", default="dallinger", help="PEM file name; default is dallinger.pem"
+)
+@click.option(
+    "--image_name",
+    default="ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20230516",
+    help="Image name; default is Ubuntu 22.04",
+)
+@click.option(
+    "--security_group_name",
+    default="dallinger",
+    help="Security group name; default is dallinger",
+)
+@click.option(
+    "--dns-host",
+    help="DNS name to use. Must resolve all its subdomains to the IP address specified as ssh host",
+    default=None,
+)
+@click.pass_context
+def ec2__provision(
+    ctx, name, region, type, storage, pem, image_name, security_group_name, dns_host
+):
+    try:
+        from dallinger import db, experiment
+
+        config = get_instance_config()
+        exp_klass = experiment.load()
+        exp = exp_klass(db.session)
+
+        asset_storage = getattr(exp, "asset_storage", None)
+        if asset_storage is not None:
+            asset_storage = asset_storage.__class__.__name__
+
+        if asset_storage and not asset_storage == "S3Storage":
+            if not user_confirms(
+                f"Your asset storage is {asset_storage} which is not recommended for experiments which use assets. "
+                f"For experiments with assets we recommend using S3Storage instead. "
+                f"Are you sure you want to provision an EC2 instance?"
+            ):
+                exit()
+    except ImportError:
+        if not user_confirms(
+            "You are not inside an experiment folder. This is not recommended. "
```

Comment:
This seems not needed, I think this only comes because of Pol making psynet specific tests that anyhow should not be there...